pkg/cfn/builder/nodegroup.go

```diff
@@ -107,33 +107,42 @@ func (n *NodeGroupResourceSet) newResource(name string, resource interface{}) *g
 }
 
 func (n *NodeGroupResourceSet) addResourcesForNodeGroup() error {
-	lc := &gfn.AWSAutoScalingLaunchConfiguration{
-		IamInstanceProfile: n.instanceProfile,
-		SecurityGroups:     n.securityGroups,
-		ImageId:            gfn.NewString(n.spec.AMI),
-		InstanceType:       gfn.NewString(n.spec.InstanceType),
-		UserData:           n.userData,
+	launchTemplateName := gfn.MakeFnSubString(fmt.Sprintf("${%s}", gfn.StackName))
+	launchTemplateData := &gfn.AWSEC2LaunchTemplate_LaunchTemplateData{
+		IamInstanceProfile: &gfn.AWSEC2LaunchTemplate_IamInstanceProfile{
+			Arn: gfn.MakeFnGetAttString("NodeInstanceProfile.Arn"),
+		},
+		SecurityGroups: n.securityGroups,
+		ImageId:        gfn.NewString(n.spec.AMI),
+		InstanceType:   gfn.NewString(n.spec.InstanceType),
+		UserData:       n.userData,
 	}
+
 	if api.IsEnabled(n.spec.SSH.Allow) && api.IsSetAndNonEmptyString(n.spec.SSH.PublicKeyName) {
-		lc.KeyName = gfn.NewString(*n.spec.SSH.PublicKeyName)
+		launchTemplateData.KeyName = gfn.NewString(*n.spec.SSH.PublicKeyName)
 	}
 	if n.spec.PrivateNetworking {
-		lc.AssociatePublicIpAddress = gfn.False()
+		launchTemplateData.NetworkInterfaces = []gfn.AWSEC2LaunchTemplate_NetworkInterface{{
```

Comment:
So it looks like we shouldn't actually define it here at all.

If so, where is this this parameter supposed to be set?

There are few things in [the docs](https://docs.aws.amazon.com/autoscaling/ec2/userguide/create-launch-template.html), but it's still not clear to me what I'm supposed to do here really.

Reply:
Or do I need to set actual `DeviceIndex` parameter?

Reply:
Looks like you might need to set Device to eth0 ?

Reply:
I tried with 
```
launchTemplateData.NetworkInterfaces = []gfn.AWSEC2LaunchTemplate_NetworkInterface{{
	AssociatePublicIpAddress: gfn.False(),
	DeviceIndex:              gfn.NewInteger(0),
}}
```

but I still got the same error:
```[ℹ]  AWS::CloudFormation::Stack/eksctl-martina-test-cluster-6-nodegroup-ng-efc71c9c: ROLLBACK_IN_PROGRESS – "The following resource(s) failed to create: [NodeGroup]. . Rollback requested by user."
[✖]  AWS::AutoScaling::AutoScalingGroup/NodeGroup: CREATE_FAILED – "Incompatible launch template: The network interface's device index must be zero. (Service: AmazonAutoScaling; Status Code: 400; Error Code: InvalidQueryParameter; Request ID: e39d1400-6b18-11e9-aa8c-a7959d188056)"```

Reply:
@christopherhein I don't think there is `Device`, it's `DeviceIndex`...

@martina-if and you've tried that with `privateNetworking: true`?

I actually managed to get past it now, and seeing this:
```
[✖]  AWS::AutoScaling::AutoScalingGroup/NodeGroup: CREATE_FAILED – "Invalid launch template: When a network interface is provided, the security groups must be a part of it. (Service: AmazonAutoScaling; Status Code: 400; Error Code: InvalidQueryParameter; Request ID: 76da1253-6b1c-11e9-a368-39534edcaa0d)"
```

Reply:
Looks like we have a working thing now, let's come back to this after 0.1.30 is out.